collector/library/pod.js

```diff
@@ -56,51 +46,44 @@ class Pod {
     }
 
     async checkAndProcessNewBatches() {
-        if (this.dead === true) {
-            return;
-        }
+        while (true) {
+            if (this.dead === true) {
+                throw new Error(`Pod for ${this.targetUrl} is now dead`);
+            }
 
-        console.log(`[Check & process new batches]`, this.targetUrl);
+            console.log(`[Check & process new batches]`, this.targetUrl);
 
-        let result, response;
+            const result = await fetch(this.targetUrl);
+            const response = await result.json();
 
-        try {
-            result = await fetch(this.targetUrl);
-            response = await result.json();
-        } catch (err) {
-            //console.log('failed getting available batches', err);
-            return;
-        }
-
-        if ('status' in response && response.status === 'error') {
-            //console.log('found error when fetching availble batches', response);
-            return;
-        }
+            if ('status' in response && response.status === 'error') {
+                throw new Error(`Pod for ${this.targetUrl} appears to have an error ${JSON.stringify(response)}`);
+            }
 
-        let targetBatch;
-        const batches = sortBy(response, ['id']);
+            let targetBatch = maxBy(batches, 'id');
+            const batches = sortBy(response, ['id']);
 
-        for (let i = 0; i < batches.length; i++) {
-            if (this.dead === true) {
-                return;
-            }
+            for (let i = 0; i < batches.length; i++) {
+                const batch = batches[i];
+                const bytesSentFromBatch = await this.getBytesSentFromThisParticularBatch({ batch });
 
-            const batch = batches[i];
-            const bytesSentFromBatch = await this.getBytesSentFromThisParticularBatch({ batch });
+                if (bytesSentFromBatch > batch.batchSize) {
+                    const errorMsg = `We collected more bytes than availble! ${this.targetUrl}, batch.id: ${batch.id}, bytes handled: ${bytesSentFromBatch}, batch.size: ${batch.batchSize}`;
+                    console.log(errorMsg);
+                    throw new Error(errorMsg);
+                }
 
-            if (bytesSentFromBatch > batch.batchSize) {
-                console.log('We collected more bytes than availble!', this.targetUrl, 'batch id:', batch.id, 'bytes handled:', bytesSentFromBatch, 'batch size:', batch.batchSize);
-                continue;
+                if (batch.batchSize > bytesSentFromBatch) {
+                    targetBatch = batch;
+                    break;
+                }
             }
 
-            if (batch.batchSize > bytesSentFromBatch) {
-                targetBatch = batch;
-                
-                break;
+            if (!targetBatch) {
```

Comment:
It may be better to replace this condition with a condition in line 62 - because you should only reach an undefined in case there are no batches at all. and its nicer to put all the assumptions checks in one place. Also. if you put it above the condition is made more readable: `batches.length === 0`

If you choose to keep this condition here please change the condition to check that `targetBatch === undefined` because you should not assume here that we never use batch id `0`